examples/deadline/EC2-Image-Builder/ts/lib/deadline-machine-image.ts

```diff
@@ -0,0 +1,198 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+
+import { IMachineImage } from '@aws-cdk/aws-ec2';
+import {
+  CfnInstanceProfile,
+  ManagedPolicy,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from '@aws-cdk/aws-iam';
+import {
+  CfnComponent,
+  CfnDistributionConfiguration,
+  CfnImage,
+  CfnImageRecipe,
+  CfnInfrastructureConfiguration,
+} from '@aws-cdk/aws-imagebuilder';
+import {
+  CfnResource,
+  Construct,
+  Token,
+} from '@aws-cdk/core';
+
+import { templateComponent } from './template';
+
+export enum OSType {
+  // The Windows operating system
+  WINDOWS = 'Windows',
+
+  // The Linux operating system
+  LINUX = 'Linux',
+}
+
+export interface DeadlineMachineImageProps {
+  /**
+   * The version of Deadline to install on the image
+   */
+  readonly deadlineVersion: string,
+
+  /**
+   * The operating system of the image.
+   */
+  readonly osType: OSType,
+
+  /**
+   * The parent image of the image recipe. Can use static methods on MachineImage to find your AMI. See
+   * https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.MachineImage.html for more details.
+   */
+  readonly parentAmi: IMachineImage,
+
+  /**
+   * The image version must be bumped any time the image or any components are modified, otherwise
+   * CloudFormation will fail to update.
+   * Must be in the format x.x.x
+   */
+  readonly imageVersion: string,
+
+  /**
+   * Customer defined Image Builder components
+   *
+   * @default - No extra components
+   */
+  readonly components?: CfnComponent[],
+
+  /**
+   * The Image Builder distribution configuration.
+   * See https://docs.aws.amazon.com/imagebuilder/latest/userguide/manage-distribution-settings.html for more info.
+   *
+   * @default - None
+   */
+  readonly distributionConfiguration?: CfnDistributionConfiguration,
+
+  /**
+   * The Image Builder infrastructure configuration.
+   * See https://docs.aws.amazon.com/imagebuilder/latest/userguide/manage-infra-config.html for more info.
+   *
+   * @default - Creates a role with the minimum required permissions to create an AMI using the Deadline component.
+   */
+  readonly infrastructureConfiguration?: CfnInfrastructureConfiguration,
+}
+
+/**
+ * Construct to setup all the required Image Builder constructs to create an AMI with Deadline installed.
+ */
+export class DeadlineMachineImage extends Construct {
+  public readonly amiId: string;
+
+  constructor(scope: Construct, id: string, props: DeadlineMachineImageProps) {
+    super(scope, id);
+
+    const infrastructureConfiguration = props.infrastructureConfiguration ?? this.createDefaultInfrastructureConfig(id);
+
+    // Create the Deadline component that will install Deadline onto any base image
+    const deadlineComponentData = this.getDeadlineComponent(
+      props.deadlineVersion,
+      props.osType,
+    );
+
+    const deadlineComponent = new CfnComponent(scope, `DeadlineComponent${id}`, {
+      platform: props.osType,
```

Comment:
Can we get this property from the parent image? Something like this:
```suggestion
      platform: props.parentAmi.getImage(this).osType,
```

Reply:
Updated